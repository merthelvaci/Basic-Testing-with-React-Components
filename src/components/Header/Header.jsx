import React from 'react';
import PropTypes from 'prop-types';

const Header = ({ header }) => (
	<div>
		<h1 data-testid="h1tag">{header}</h1>
	</div>
);

Header.propTypes = {
	header: PropTypes.string.isRequired,
};

export default Header;
