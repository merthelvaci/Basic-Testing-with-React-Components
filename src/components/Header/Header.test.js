import Header from '.';

afterEach(cleanup);

describe('<Header /> component tests', () => {
	const testHeader = 'My Header';
	it('should render Header component correctly', () => {
		const { asFragment } = render(<Header header={testHeader} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it(`should have h1 element with text content ${testHeader}`, () => {
		const { getByTestId } = render(<Header header={testHeader} />);
		expect(getByTestId('h1tag')).toHaveTextContent(testHeader);
	});
});
