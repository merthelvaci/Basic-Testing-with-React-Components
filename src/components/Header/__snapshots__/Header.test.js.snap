// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component tests should render Header component correctly 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="h1tag"
    >
      My Header
    </h1>
  </div>
</DocumentFragment>
`;
